include/range/v3/numeric/partial_sum.hpp

```diff
@@ -30,16 +30,14 @@ namespace ranges
     inline namespace v3
     {
         template<typename I, typename O, typename BOp = plus, typename P = ident,
-            typename V = value_type_t<I>,
-            typename X = concepts::Invocable::result_t<P&, V>,
-            typename Y = concepts::Invocable::result_t<BOp&, X, X>>
+            typename X = projected<projected<I, coerce<value_type_t<I>>>, P>,
+            typename Y = reference_t<X>>
         using PartialSummable = meta::strict_and<
+            SemiRegular<detail::decay_t<Y>>,
```

Comment:
Why not make it `Regular`? Or `Regular && StreamInsertable`? Constraining the arguments beyond the requirements of the algorithm is counterproductive. This implementation does four things with `detail::decay_t<Y>`:
* constructs an object of that type from a `Y`
* writes an lvalue through an `O`
* assigns an lvalue from the result of invoking an lvalue `BOp` with /*...*/
* destroys the object

I cannot envision a more efficient implementation of this algorithm that would make use of `SemiRegular<detail::decay_t<Y>>`.